pkg/target/target.go

```diff
@@ -444,41 +459,83 @@ func convertToMatch(object map[string]interface{}) (*match.Match, error) {
 func (h *K8sValidationTarget) ToMatcher(u *unstructured.Unstructured) (constraints.Matcher, error) {
 	obj, found, err := unstructured.NestedMap(u.Object, "spec", "match")
 	if err != nil {
-		return nil, fmt.Errorf("%w: %v", ErrCreatingMather, err)
+		return nil, fmt.Errorf("%w: %v", ErrCreatingMatcher, err)
 	}
 	if found && obj != nil {
 		match, err := convertToMatch(obj)
 		if err != nil {
-			return nil, fmt.Errorf("%w: %v", ErrCreatingMather, err)
+			return nil, fmt.Errorf("%w: %v", ErrCreatingMatcher, err)
 		}
-		return &Matcher{match}, nil
+		return &Matcher{match, &h.cache}, nil
 	}
-	return nil, fmt.Errorf("%w: %s %s has no match found", ErrCreatingMather, u.GetKind(), u.GetName())
+	return nil, fmt.Errorf("%w: %s %s has no match found", ErrCreatingMatcher, u.GetKind(), u.GetName())
 }
 
 // Matcher implements constraint.Matcher.
-// TODO: add cache when handler.Cache interface is available.
 type Matcher struct {
 	match *match.Match
+	cache *nsCache
+}
+
+type nsCache struct {
```

Comment:
do we want to replace https://github.com/open-policy-agent/gatekeeper/blob/a3d8a0dd9b2d94f3d4a28fc11c7f5e43e1d15e64/pkg/audit/manager.go#L102 this with frameworks implementation too?

Reply:
What do you mean by "make this generic"? Indicate that we should be able to cache other objects?

If so, we could, but we probably don't want to do that currently since we'd be duplicating OPA's cache. It's something we may want to consider as possible future work if we needed wider access to the cache, though.

Reply:
*follow-up = after merging in the frameworks changes

Reply:
SGTM as a follow-up